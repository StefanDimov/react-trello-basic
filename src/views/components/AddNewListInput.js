import React from 'react'
import { FormGroup, FormControl } from 'react-bootstrap'

export default class AddNewListInput extends React.Component {
    constructor(props) {
        super(props)

        this.state = { value: '' }

        this.changeValueHandler = this.changeValueHandler.bind(this)
        this.onSubmit = this.onSubmit.bind(this)
    }

    changeValueHandler(e) {
        this.setState({ value: e.target.value })
    }

    onSubmit(e) {
        e.preventDefault();
        this.props.onCreateList(this.state.value)
        this.setState({ value: '' })
    }

    render() {
        return (
            <form className="my-list-container-item" onSubmit={this.onSubmit}>
                <FormGroup>
                    <FormControl
                        type="text"
                        value={this.state.value}
                        onChange={this.changeValueHandler}
                        placeholder="Add new list..." />
                </FormGroup>
            </form>
        )
    }
}

const { func } = React.PropTypes
AddNewListInput.propTypes = {
    onCreateList: func.isRequired
}
