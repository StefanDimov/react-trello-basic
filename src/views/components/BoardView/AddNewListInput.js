import React from 'react'
import { FormGroup, FormControl } from 'react-bootstrap'

/**
 * Component with an input for adding a new list. Meant to be used in Board Component.
 * @class comp/AddNewListInput
 * @param {function} onCreateList callback for input submit
 */
export default class AddNewListInput extends React.Component {
    constructor(props) { // eslint-disable-line
        super(props)

        this.state = { value: '' }

        this.changeValueHandler = this.changeValueHandler.bind(this)
        this.onSubmit = this.onSubmit.bind(this)
    }

    /**
     * Controls the input change event
     * @private
     * @param {object} e event object
     */
    changeValueHandler(e) {
        this.setState({ value: e.target.value })
    }

    /**
     *  Controls the form submit event
     *  @private
     *  @param {object} e event object
     */
    onSubmit(e) {
        e.preventDefault()
        this.props.onCreateList(this.state.value)
        this.setState({ value: '' })
    }

    render() { // eslint-disable-line
        return (
            <form className="my-list-container-item" onSubmit={this.onSubmit}>
                <FormGroup>
                    <FormControl
                        type="text"
                        value={this.state.value}
                        onChange={this.changeValueHandler}
                        placeholder="Add new list..." />
                </FormGroup>
            </form>
        )
    }
}

const { func } = React.PropTypes
AddNewListInput.propTypes = {
    onCreateList: func.isRequired
}
