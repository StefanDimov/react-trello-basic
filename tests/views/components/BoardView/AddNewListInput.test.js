import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import AddNewListInput from '../../../../src/views/components/BoardView/AddNewListInput'
import { FormControl } from 'react-bootstrap'

describe('AddNewListInput', () => {

    const inputChangeEventDataWithName = { target: { value: 'list name' } }
    const onCreateList = jest.fn()

    beforeEach(() => {
        onCreateList.mockClear()
    })

    it('should render correctly', () => {
        const tree = renderer.create(
            <AddNewListInput onCreateList={onCreateList} />
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('should call onCreateList with the new list\'s name on submit', () => {
        const eventMock = { preventDefault: jest.fn() }

        const wrapper = shallow(<AddNewListInput onCreateList={onCreateList} />)

        wrapper.find(FormControl).simulate('change', inputChangeEventDataWithName)
        wrapper.simulate('submit', eventMock)

        expect(onCreateList).toHaveBeenLastCalledWith(inputChangeEventDataWithName.target.value)
        expect(eventMock.preventDefault).toHaveBeenCalled()
    })

    it('should update state on change', () => {
        const wrapper = shallow(<AddNewListInput onCreateList={onCreateList} />)
        wrapper.find(FormControl).simulate('change', inputChangeEventDataWithName)
        expect(wrapper.find(FormControl).prop('value')).toBe(inputChangeEventDataWithName.target.value)
    })
})
