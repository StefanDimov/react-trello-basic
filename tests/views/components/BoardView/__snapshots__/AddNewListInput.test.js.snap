exports[`AddNewListInput should render correctly 1`] = `
<form
  className="my-list-container-item"
  onSubmit={[Function]}>
  <div
    className="form-group">
    <input
      className="form-control"
      id={undefined}
      onChange={[Function]}
      placeholder="Add new list..."
      type="text"
      value="" />
  </div>
</form>
`;
